src/libraries/System.Private.CoreLib/src/System/Byte.cs

```diff
@@ -382,11 +382,11 @@ static byte IBinaryNumber<byte>.Log2(byte value)
 
         [RequiresPreviewFeatures]
         static byte IDecrementOperators<byte>.operator --(byte value)
-            => value--;
+            => --value;
```

Comment:
Is this just for consistency?

Reply:
Not quite, this is mostly for correctness.

`value--` is the same as:
```csharp
var tmp = value;
value -= 1;
return tmp;
```

So we need to do either `=> value - 1` or `--value`, and I opted for using the operator that we are exposing for consistency.

Reply:
Ugh, yes, I wasn't thinking.

Reply:
That said, `value - 1` and `--value` produce different IL, and at least according to sharplab, slightly different asm, with the former being smaller/better.  That's probably worth more than consistency (unless we can improve the compiler here instead).
https://sharplab.io/#v2:EYLgxg9gTgpgtADwGwBYA0AXEBLANgHwAEAmARgFgAoQgZgAIS6BhOgbyrs4fuAE8MYdALKkAFHwF1gASjoBeAHx1x/GNPF04dUtIDcHLrSmrhxFZJnylcOMH2UAvkA=
https://sharplab.io/#v2:EYLgxg9gTgpgtADwGwBYA0AXEBDAzgWwB8ABAJgEYBYAKGIGYACMhgYQYG8aHunHgBPDDAYBZcgAoBQhsACUDALwA+BpMExZkhnAblZAbi496M9aNJrpcxSrhxgh6gF8gA==

Reply:
That secondary zero extension should be elided in .NET 6 (but I'll need to double check).

I'll log an issue to follow up on it.

Reply:
Although, it might be worth doing `b - 1` for the smaller IL though even if the secondary zero extension is removed...

Reply:
@tannergooding while it will be elided in this scenario, different register allocation is likely to interfere with that. Please file an issue and assign it to me. I think we'll be able to get to only one extension for the whole method.

Reply:
Fixed in a follow up PR.